gpii/node_modules/accessRequester/test/AccessRequesterTests.js

```diff
@@ -0,0 +1,226 @@
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    kettle = require("kettle"),
+    nock = require("nock");
+
+require("accessRequester");
+// To reuse gpii.tests.oauth2.invokePromiseProducer()
+fluid.require("%universal/gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/test/js/DataStoreTestsUtils.js");
+
+kettle.loadTestingSupport();
+
+fluid.registerNamespace("gpii.tests.accessRequester");
+
+gpii.tests.accessRequester.hostname = "http://gpii.net";
+gpii.tests.accessRequester.url = "/access_token";
+gpii.tests.accessRequester.gpiiToken = "gpii-token-test";
+
+// Set up mock ajax responses
+gpii.tests.accessRequester.setupNock = function (config) {
+    var cloudMock = nock(gpii.tests.accessRequester.hostname);
+
+    // log nock matches
+    cloudMock.log(console.log);
+
+    // mock POST requests to "/access_token"
+    cloudMock.post(gpii.tests.accessRequester.url, config.data)
+        .reply(config.status, config.response);
+};
+
+// The customized accessRequester component for tests
+fluid.defaults("gpii.tests.accessRequester", {
+    gradeNames: ["gpii.accessRequester"],
+    gpiiToken: gpii.tests.accessRequester.gpiiToken,
+    url: gpii.tests.accessRequester.hostname + gpii.tests.accessRequester.url,
+    clientCredentialDataSourceGrade: "gpii.accessRequester.clientCredentialDataSource.file"
+});
+
+// The base testEnvironment grade to be inherited by all tests
+fluid.defaults("gpii.tests.accessRequesterTests", {
+    gradeNames: ["fluid.test.testEnvironment"],
+    clientCredentialFilePath: null,    // supplied by individual tests
+    testCaseHolderGrade: null,   // supplied by individual tests
+    distributeOptions: {
+        clientCredentialFilePath: {
+            source: "{that}.options.clientCredentialFilePath",
+            target: "{that clientCredentialDataSource}.options.path"
+        },
+        testCaseHolderGrade: {
+            source: "{that}.options.testCaseHolderGrade",
+            target: "{that > testCaseHolder}.type"
+        }
+    },
+    components: {
+        accessRequester: {
+            type: "gpii.tests.accessRequester"
+        },
+        testCaseHolder: {
+            type: "fluid.test.testCaseHolder"
+        }
+    }
+});
+
+// The base testCaseHolder grade to be inherited by test case holders used by all tests
+fluid.defaults("gpii.tests.accessRequester.testCaseHolder", {
+    gradeNames: "fluid.test.testCaseHolder",
+    events: {
+        onResponse: null,
+        onError: null
+    }
+});
+
+// 1. A successful workflow
+gpii.tests.accessRequester.success = {
+    clientCredentialFilePath: "%universal/gpii/node_modules/accessRequester/test/data/clientCredential-correct.json",
+    nockConfig: {
+        data: {
```

Comment:
Am I right to think that this is the request data to match against? If so, I would rename this from `data` to `request`.